python/vyos/ifconfig/interface.py

```diff
@@ -373,14 +406,21 @@ def _delete(self):
         # NOTE (Improvement):
         # after interface removal no other commands should be allowed
         # to be called and instead should raise an Exception:
-        cmd = 'ip link del dev {ifname}'.format(**self.config)
+        netns = get_interface_namespace(self.config['ifname'])
```

Comment:
Why not use the netns name from vyos CLI config by accessing self.config()?

Reply:
For delete, we can't get data from `self.config{'netns'[`
It will be empty config

Reply:
Thanks